src/bech32.cpp

```diff
@@ -161,26 +155,28 @@ std::pair<std::string, data> Decode(const std::string& str) {
     for (size_t i = 0; i < str.size(); ++i) {
         unsigned char c = str[i];
         if (c < 33 || c > 126) return {};
-        if (c >= 'a' && c <= 'z') lower = true;
-        if (c >= 'A' && c <= 'Z') upper = true;
+        // Mixing upper and lowercase is not OK.
+        if (!lower && c >= 'a' && c <= 'z') { if(upper) return {}; lower = true; }
```

Comment:
Nit: `if (upper)` instead of `if(upper)`

Reply:
Not even a nit. Should be a compiler error IMO. I must be tired. :-)